02_programming_fundamentals/03_algorithms_day_1/05_play_with_for_loop_bonus/.__tests__/04_inverted_right_triangle.test.js

```diff
@@ -1,35 +1,27 @@
 /* global beforeAll test expect */
 const fs = require("fs");
 const path = require("path");
+const readcode = require("../../../.test_utils/readcode");
 
 let studentCode;
-
 beforeAll(() => {
-  // Loads the content of the student's code
-  return new Promise(function(resolve, reject) {
-    fs.readFile(
-      path.join(__dirname, "../04_inverted_right_triangle.js"),
-      "utf8",
-      function(err, text) {
-        if (err) {
-          reject(err);
-          return;
-        }
-        studentCode = text;
-        resolve();
-      }
-    );
-  });
+  // Loads the student's code
+  studentCode = readcode(
+    path.resolve(__dirname, "../04_inverted_right_triangle.js")
+  );
+  return studentCode;
 });
 
 test("display an inverted right triangle", () => {
-  output = [];
-  _consolelog = console.log;
-  console.log = thing => output.push(thing);
+  return studentCode.then(code => {
+    output = [];
+    _consolelog = console.log;
+    console.log = thing => output.push(thing);
```

Comment:
Is this relevant to make a `collectOutput` function? 🤔

Reply:
I think there is a way of doing that cleanly with Jest but I'll do another PR